ehitchhiking-ui/src/app/shared/modals/profile-modal/profile-modal.component.ts

```diff
@@ -42,22 +43,45 @@ export class ProfileModalComponent implements OnInit {
 		this.addCarMod = !this.addCarMod;
 	}
 
-	onSubmitNewCar(newCar: Car): void {
-		this.user.addCar(newCar);
-		this.carsInfoForm = this.carsInfoService.toFormGroup(this.user.cars, this.formBuilder);
-		this.addCarMod = false;
+	onSubmitNewCar(newCar: any): void {
+		this.isLoading = true;
+		this.profileModalService.addNewCar(newCar, this.currentUser.id).then((car) => {
+			if (car) {
+				this.currentUser.addCar(car);
+				this.carsInfoForm = this.carsInfoService.toFormGroup(this.currentUser.cars, this.formBuilder);
+			}
+			this.addCarMod = false;
+			this.isLoading = false;
+		});
 	}
 
 	onSubmitCarsChanges(): void {
-		const newCars = this.carsInfoService.getCarsInfo(this.carsInfoForm, this.user.cars.length);
-		this.user.cars = newCars;
+		const newCars = this.carsInfoService.getCarsInfo(
+			this.carsInfoForm,
+			this.currentUser.cars.length,
+			this.currentUser.cars
+		);
+		this.isLoading = true;
+		this.profileModalService.updateCars(newCars, this.currentUser.id).then((response) => {
+			if (response) {
+				this.currentUser.cars = response;
+				this.carsInfoForm = this.carsInfoService.toFormGroup(response, this.formBuilder);
+				this.isLoading = false;
+			}
+		});
```

Comment:
pls use finally block to disable loading

Reply:
made it